### ISSUE TITLE
Failed to create custom docker image: windows_64_shared_msvc_512

### ISSUE THREAD

**Issue by akiyosi** [therecipe/qt#1024]
I want to create a custom Docker image for Windows that can build MSVC using Qt5.12.6.
I prepared a Dockerfile and related files as below with reference to [link](https://github.com/therecipe/qt/blob/master/internal/docker/wine/Dockerfile.64_msvc).

https://github.com/akiyosi/goneovim/tree/add-build-scripts/build/windows_64_shared_mcvc_512

However, I tried to build, but the following problems occur.

https://gist.github.com/akiyosi/b1133533826030e16c39a9381c730ff6#file-error-log

If comment-out [L62](https://github.com/akiyosi/goneovim/blob/add-build-scripts/build/windows_64_shared_mcvc_512/Dockerfile.64_msvc#L62), it is success to create the custom image, but it seems to fail with a similar problem when building with my project.
On the other hand, if I use Dockerhub's MSVC container image `windows_64_shared_msvc`, the build will be successful.

**Comment by akiyosi** [comment]
I also tried an MSVC build on windows10. However, the following error occurs:

```
time="2019-12-04T17:07:42+09:00" level=info msg="running: 'qtsetup prep'"
time="2019-12-04T17:07:42+09:00" level=info msg="successfully created qtrcc symlink in your PATH (D:\\msys2\\usr\\local\\bin\\go\\bin\\qtrcc.exe)"
time="2019-12-04T17:07:42+09:00" level=info msg="successfully created qtmoc symlink in your PATH (D:\\msys2\\usr\\local\\bin\\go\\bin\\qtmoc.exe)"
time="2019-12-04T17:07:42+09:00" level=info msg="successfully created qtminimal symlink in your PATH (D:\\msys2\\usr\\local\\bin\\go\\bin\\qtminimal.exe)"
time="2019-12-04T17:07:42+09:00" level=info msg="successfully created qtdeploy symlink in your PATH (D:\\msys2\\usr\\local\\bin\\go\\bin\\qtdeploy.exe)"
time="2019-12-04T17:09:40+09:00" level=info msg="running: 'qtsetup check windows' [docker=false] [vagrant=false]"
time="2019-12-04T17:09:42+09:00" level=info msg="GOOS:                        'windows'"
time="2019-12-04T17:09:42+09:00" level=info msg="GOARCH:                      'amd64'"
time="2019-12-04T17:09:42+09:00" level=info msg="GOVERSION:                   'go1.10'"
time="2019-12-04T17:09:42+09:00" level=info msg="GOROOT:                   *  'D:/msys2/usr/local/bin/go'"
time="2019-12-04T17:09:42+09:00" level=info msg="GOPATH:                   *  'D:/msys2/home/akiyosi/gopath/'"
time="2019-12-04T17:09:42+09:00" level=info msg="GOBIN:                       'D:\\msys2\\home\\akiyosi\\gopath\\bin'"
time="2019-12-04T17:09:42+09:00" level=info msg="GOMOD:                       ''"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_HASH:                     '47559b8cfa23e0cb27bbf8aa2f654b5c9922a849'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_API:                      ''"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_VERSION:               *  '5.12.6'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_DIR:                   *  'D:\\msys2\\usr\\local\\windowsApps\\Qt5.12.6'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_STUB:                     'false'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_DEBUG:                    'false'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_QMAKE_DIR:                ''"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_WEBKIT:                   'false'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_DEBUG_CONSOLE:            'false'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_MSYS2:                    'false'"
time="2019-12-04T17:09:42+09:00" level=info msg="QT_MSVC:                  *  'true'"
time="2019-12-04T17:09:42+09:00" level=info msg="GOVSVARSPATH:             *  'D:/msys2/usr/local/windowsApps/Microsoft Visual Stud
io/2017/Professional/VC/Auxiliary/Build/vcvars64.bat'"
time="2019-12-04T17:09:42+09:00" level=info msg="running: 'qtsetup generate windows' [docker=false] [vagrant=false]"
time="2019-12-04T17:09:49+09:00" level=info msg="generating full qt/core"
time="2019-12-04T17:10:05+09:00" level=info msg="generating full qt/androidextras"
time="2019-12-04T17:10:06+09:00" level=info msg="generating full qt/gui"
time="2019-12-04T17:10:23+09:00" level=info msg="generating full qt/network"
time="2019-12-04T17:10:27+09:00" level=info msg="generating full qt/xml"
time="2019-12-04T17:10:29+09:00" level=info msg="generating full qt/dbus"
time="2019-12-04T17:10:31+09:00" level=info msg="generating full qt/nfc"
time="2019-12-04T17:10:32+09:00" level=info msg="generating full qt/script"
time="2019-12-04T17:10:34+09:00" level=info msg="generating full qt/sensors"
time="2019-12-04T17:10:39+09:00" level=info msg="generating full qt/positioning"
time="2019-12-04T17:10:41+09:00" level=info msg="generating full qt/widgets"
time="2019-12-04T17:11:21+09:00" level=info msg="generating full qt/sql"
time="2019-12-04T17:11:24+09:00" level=info msg="generating full qt/qml"
time="2019-12-04T17:11:26+09:00" level=info msg="generating full qt/websockets"
time="2019-12-04T17:11:27+09:00" level=info msg="generating full qt/xmlpatterns"
time="2019-12-04T17:11:29+09:00" level=info msg="generating full qt/bluetooth"
time="2019-12-04T17:11:32+09:00" level=info msg="generating full qt/webchannel"
time="2019-12-04T17:11:33+09:00" level=info msg="generating full qt/svg"
time="2019-12-04T17:11:34+09:00" level=info msg="generating full qt/multimedia"
time="2019-12-04T17:11:46+09:00" level=info msg="generating full qt/quick"
time="2019-12-04T17:11:51+09:00" level=info msg="generating full qt/help"
time="2019-12-04T17:11:55+09:00" level=info msg="generating full qt/location"
time="2019-12-04T17:11:57+09:00" level=info msg="generating full qt/scripttools"
time="2019-12-04T17:11:57+09:00" level=info msg="generating full qt/uitools"
time="2019-12-04T17:11:58+09:00" level=info msg="generating full qt/winextras"
time="2019-12-04T17:12:00+09:00" level=info msg="generating full qt/webengine"
time="2019-12-04T17:12:03+09:00" level=info msg="generating full qt/testlib"
time="2019-12-04T17:12:04+09:00" level=info msg="generating full qt/serialport"
time="2019-12-04T17:12:05+09:00" level=info msg="generating full qt/serialbus"
time="2019-12-04T17:12:08+09:00" level=info msg="generating full qt/printsupport"
time="2019-12-04T17:12:11+09:00" level=info msg="generating full qt/designer"
time="2019-12-04T17:12:15+09:00" level=info msg="generating full qt/scxml"
time="2019-12-04T17:12:17+09:00" level=info msg="generating full qt/gamepad"
time="2019-12-04T17:12:18+09:00" level=info msg="generating full qt/purchasing"
time="2019-12-04T17:12:19+09:00" level=info msg="generating full qt/datavisualization    [GPLv3]"
time="2019-12-04T17:12:24+09:00" level=info msg="generating full qt/charts               [GPLv3]"
time="2019-12-04T17:12:34+09:00" level=info msg="generating full qt/virtualkeyboard      [GPLv3]"
time="2019-12-04T17:12:35+09:00" level=info msg="generating full qt/speech"
time="2019-12-04T17:12:36+09:00" level=info msg="generating full qt/quickcontrols2"
time="2019-12-04T17:12:37+09:00" level=info msg="generating full qt/sailfish"
time="2019-12-04T17:12:37+09:00" level=info msg="generating full qt/webview"
time="2019-12-04T17:12:37+09:00" level=info msg="generating full qt/remoteobjects"
time="2019-12-04T17:12:48+09:00" level=info msg="running: 'qtsetup install windows' [docker=false] [vagrant=false]"
time="2019-12-04T17:12:48+09:00" level=info msg="installing full qt/core"
runtime/cgo
# runtime/cgo
Microsoft(R) C/C++ Optimizing Compiler Version 19.16.27034 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

cl : �R�}���h ���C�� error D8021 : ���l�^���� '/Werror' �͖����ł��B
```

**Comment by therecipe** [comment]
Hey

Sorry for the wait.
I looked into this and it seems like this was caused by using the `staging` versions of wine.

I fixed this now with https://github.com/therecipe/qt/commit/5e239f03fa53f391115b235bf47c205c8a633471, and to fix your docker file, just replace `staging` with `stable` here https://github.com/akiyosi/goneovim/blob/add-build-scripts/build/windows_64_shared_mcvc_512/Dockerfile.64_msvc#L48

**Comment by akiyosi** [comment]
I appreciate your support.
Following your guidance, I have successfully created a container image.

However, I fail to build my project.
The error is very similar to the one reported in the #1025.
The following errors are errors when building the example of [goqtframelesswindow](https://github.com/akiyosi/goqtframelesswindow) project using [akiyosi/qt:windows_64_shared_msvc_512](https://hub.docker.com/repository/docker/akiyosi/qt).

https://gist.github.com/akiyosi/22726d1fda0e33332ff6a90fccb2a464

Could this still involve `qt_api`? I will create a container without the qt_api setting and try to build it.

By the way, the issue has been solved, so I will close it.
